core/store/src/trie/state_snapshot.rs

```diff
@@ -153,52 +153,39 @@ impl ShardTries {
     }
 
     /// Makes a snapshot of the current state of the DB.
-    /// If a snapshot was previously available, it gets deleted.
-    pub fn make_state_snapshot(
+    pub fn create_state_snapshot(
         &self,
-        prev_block_hash: &CryptoHash,
+        prev_block_hash: CryptoHash,
         shard_uids: &[ShardUId],
         block: &Block,
     ) -> Result<(), anyhow::Error> {
         metrics::HAS_STATE_SNAPSHOT.set(0);
         // The function returns an `anyhow::Error`, because no special handling of errors is done yet. The errors are logged and ignored.
         let _span =
-            tracing::info_span!(target: "state_snapshot", "make_state_snapshot", ?prev_block_hash)
+            tracing::info_span!(target: "state_snapshot", "create_state_snapshot", ?prev_block_hash)
                 .entered();
-        tracing::info!(target: "state_snapshot", ?prev_block_hash, "make_state_snapshot");
-
-        let StateSnapshotConfig { home_dir, hot_store_path, state_snapshot_subdir, .. } =
-            self.state_snapshot_config();
+        let _timer = metrics::CREATE_STATE_SNAPSHOT_ELAPSED.start_timer();
 
-        let _timer = metrics::MAKE_STATE_SNAPSHOT_ELAPSED.start_timer();
         // `write()` lock is held for the whole duration of this function.
-        // Accessing the snapshot in other parts of the system will fail.
-        let mut state_snapshot_lock = self.state_snapshot().write().map_err(|err| {
-            anyhow::Error::msg(format!(
-                "error accessing write lock of state_snapshot: {}",
-                err.to_string()
-            ))
-        })?;
+        let mut state_snapshot_lock = self.state_snapshot().write().unwrap();
         let db_snapshot_hash = self.get_state_snapshot_hash();
-
         if let Some(state_snapshot) = &*state_snapshot_lock {
             // only return Ok() when the hash stored in STATE_SNAPSHOT_KEY and in state_snapshot_lock and prev_block_hash are the same
-            if db_snapshot_hash.is_ok()
-                && db_snapshot_hash.unwrap() == *prev_block_hash
-                && state_snapshot.prev_block_hash == *prev_block_hash
+            if db_snapshot_hash.is_ok_and(|hash| hash == prev_block_hash)
```

Comment:
@wacban earlier had the comment to use `db_snapshot_hash == Ok(prev_block_hash)`. That doens't work

Reply:
(Fine to keep as is)
I'm surprised and ChatGPT can't explain the difference to me.